nxviz/plots.py

```diff
@@ -153,7 +153,12 @@ def __init__(
         figsize = (6, 6)
         if "figsize" in kwargs.keys():
             figsize = kwargs["figsize"]
-        self.figure = plt.figure(figsize=figsize)
+        if "fig" in kwargs.keys():
```

Comment:
The API you've proposed here uses `**kwargs`. I'm cool with the idea, but I also want to be thoughtful about the API. nxviz's API is its biggest 'selling point', so I put more oversight on the API than I do on features. Code maintainability is important too, though it only takes a slight backseat to the API.

I was wondering if we could do an API design thought experiment here. 

For the use case where we are plotting a CircosPlot to an existing figure, which one looks better to you?

**Option 1, which uses idioms from the PyData ecosystem**

```python
# Pre-instantiate a figure and axes object.
fig = plt.figure()
ax = fig.add_subplot(111)
c = CircosPlot(G, ..., ax=ax)
```

**Option 2, your current proposed API**

```python
c = CircosPlot(G, ..., fig=30, sub=(1,1,1)
```

**Option 3, a "fig_kwargs" API**

```python
fig_kwargs = dict(
    figsize=(10, 10),
    subplots=(1,1,1)
)
c = CircosPlot(G, ..., fig_kwargs=fig_kwargs)
```

Reply:
Of the three, I personally lean most towards option 1, which plays nicely with how people use other Python packages. Of the remaining two, I have no strong opinions. If I were to weight the three options, out of 100 points, I would assign 70-15-15. 

What are your thoughts?

Reply:
I agree, 1 is certainly better than 2 (but requires more changes?). 3 could be an option if one would plan to later implement more options and have them nicely bundled. So: 70-10-20.

Reply:
@ahoc I think option 1 only requires an isolated conditional: if `ax` is not specified, then make a figure and an axes object. In this case, `ax` should be `None` by default.

Reply:
right